_posts/2018/2018-01-07-extract-conditionals-into-well-named-methods.md

```diff
@@ -0,0 +1,101 @@
+---
+title: 'Extract conditionals into well-named methods'
+description: 'An extraction to a private method often adds clarity'
+layout: article
+category: ruby
+image:
+  base: '2018/extract-conditionals-into-well-named-methods'
+  alt: 'Jigsaw pieces'
+  source: 'https://unsplash.com/photos/3y1zF4hIPCg'
+  credit: 'Hans-Peter Gauster'
+---
+
+One of the simplest (and most effective) ways to refactor is to [extract a method](https://refactoring.com/catalog/extractMethod.html). The naming of an extracted method is a great tool for communicating the thinking behind the code.
+
+## Instead of…
+
+...using slightly complex statements in your conditionals.
+
+```ruby
+class BrightonCoffeeShop
+  def initialize(name)
+    @name = name
+  end
+
+  def good?
+    if @name == 'Starbucks' || @name == 'Costa' ||
+       @name == 'Barry’s Caff'
+      false
+    elsif @name == 'Coffee@33' || @name == 'Small Batch'
+      true
+    else
+      false
+    end
+  end
+end
+```
+
+This implementation explicitly returns `false` if the `name` is unknown. The only way to guarantee a decent coffee shop experience is to rule out places you’ve never heard of.
```

Comment:
Perhaps: *After all, the only way to guarantee..."

Reply:
Kinda feels like this bit should be in parentheses maybe? Or like a footnote (still immediately after the code) with an asterisk?

This enables the "Instead of...Use.." sentence to flow better.